File: src/main/kotlin/com/itmo/microservices/demo/warehouse/impl/repository/CatalogItemRepository.java
package com.itmo.microservices.demo.warehouse.impl.repository;

import com.itmo.microservices.demo.warehouse.impl.entity.CatalogItem;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface CatalogItemRepository extends JpaRepository<CatalogItem, UUID> {
    List<CatalogItem> findAll();
    boolean existsById(UUID id);
    CatalogItem findCatalogItemById(UUID id);
}

File: src/main/kotlin/com/itmo/microservices/demo/warehouse/impl/repository/CatalogItemRepository.kt
package com.itmo.microservices.demo.warehouse.impl.repository

import com.itmo.microservices.demo.warehouse.impl.entity.CatalogItem
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface CatalogItemRepository : JpaRepository<CatalogItem?, UUID?> {
    override fun findAll(): List<CatalogItem>
    override fun existsById(id: UUID): Boolean
    fun findCatalogItemById(id: UUID?): CatalogItem?
}